modules.d/95hwdb/module-setup.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# This file is part of dracut.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+check() {
+    return 255
+}
+
+# called by dracut
+install() {
+    local hwdb_bin
+
+    # systemd-hwdb ships the file in /etc, with /usr/lib as an alternative.
+    # Therefore consider this location as preferred for configuration.
+    hwdb_bin="${udevdir}"/hwdb.bin
```

Comment:
`${udevdir}/hwdb.bin` is already unconditionally installed by `01systemd-udevd`, so it can be ignored here.

Reply:
Right, I was rather thinging on systemd-udevd depending on this module, which might handle hwdb in more complex way; also not install multiple ones unnecessarily. Please also note that the install is not optional (`-o`) intentionally; and this module is not enabled by default. Does that make sense?

There's also an additional benefit of this module working correctly without systemd-udevd :) in case that's ever needed.

Reply:
> There's also an additional benefit of this module working correctly without systemd-udevd :)

That's a benefit without any value. ATM, HWDB is configuration for `systemd-udevd`, just like `.rules` files. It _could_ be used for other things, but it's not.

Reply:
> > There's also an additional benefit of this module working correctly without systemd-udevd :) in case that's ever needed.
> 
> This makes sense to me. This situation is very similar to udev rules - they are installed both by `systemd-udevd` and `udev-rules` .

Uff. Does `udev-rules` even work on modern systems? The list of `.rules` files seem to be rather short. (OTOH, it contains stuff that makes no sense in initrd, like `70-uaccess.rules` or `71-seat.rules`.)

Reply:
I'd say if it's not breaking anything, let's keep it (and this also verifies `hwdb.bin` is present / installed, as udevd module simply runs opt-install).